### ISSUE TITLE
Links must ensure contrast / underlining to be distinguishable

### ISSUE THREAD

**Issue by marcoscv-work** [liferay/clay#5283]
## Reference information:

**WCAG 2.1** (updated)
https://www.w3.org/WAI/WCAG21/Understanding/use-of-color
https://www.w3.org/WAI/WCAG21/Techniques/failures/F73
**WCAG 2.0** (older)
https://www.w3.org/TR/WCAG20-TECHS/F73.html
https://www.w3.org/TR/WCAG20-TECHS/G183.html

All links in Clay should be reviwed to follow the new Lexicon pattern:

https://www.figma.com/file/r0oZ3ORLODdHmvvcyCASoQ/Links?node-id=1%3A56&t=C7qv8XUFgWbgUbRk-1
<img width="800" alt="image" src="https://user-images.githubusercontent.com/7963804/210556087-18b3dc30-50cf-4fd7-a78d-fcabf3141a83.png">

**How to test it**:
Check that each link in the page that is identifiable by color (hue) is visually identifiable via some other means (e.g., underlined, bolded, italicized, sufficient difference in lightness, etc).

Since in Liferay is really difficult to ensure the sourounded color because in admin controls we have really heavy text contrast we hacve decided to provide underline in all non-component contextual content.

Potentially it could be solved with general CSS, but probably better if it is fixed in components like clay-link.

There is a task in DXP to cover all what Clay won't be able to style:
https://issues.liferay.com/browse/LPS-172123

**Comment by pat270** [comment]
Can we make Primary Link - Danger just Danger Link?

**Comment by emiliano-cicero** [comment]
Hey @pat270, yes it can be a simple Danger link. 

About the hover state, I have a proposal: 
<img width="386" alt="image" src="https://user-images.githubusercontent.com/27729173/211789970-1db681c9-5781-4216-8104-0c27cf0ae214.png">

Changing the text color would be almost imperceptible, how do you feel about adding a background change as in the image above? 

Also, for the Tertiary case (and Subtle -- if you have another name for this it's ok too) we should keep only the underline hover to avoid having e lot of effects in the interface. The idea behind the Tertiary and Subtle is to use them for interface elements such as Card titles, List titles, and Page titles, this way we avoid having underlined texts for cases where it shouldn't be required.

**Comment by pat270** [comment]
@emiliano-cicero the background colors should be no problem.

**Comment by pat270** [comment]
@emiliano-cicero There is no example of links with icons. I'm working on a ticket for User System Management and they have a link that opens to new window with icon. I don't see an example on the Figma other than don't use icon for accessibility. How should we handle the link in the gif below? Should it be underlined, change background-color on hover, leave it as is?

![link-with-icon](https://user-images.githubusercontent.com/788266/219445021-15800970-4850-41ca-a051-0df7ad81c1a9.gif)

**Comment by emiliano-cicero** [comment]
hey @pat270, I think it makes sense, in this case, to keep the icon since it will help understand that the link will open in another window. Can you please make sure the spacing between the link and the icon is 8px? 

Regarding the style, I think we will need to add the underline also for the default state to make it consistent with the other links, and for the hover it should also be the same as the other links without icons.
